general:
  height: 260                       # [px]
  width: 346                        # [px]
  undistort: true
  line_publish_rate: 40
  periodical_check_frequency: 60    # [hz]
  write_for_visualization: false

hw_test:
  mask_height: 260                  # [px]
  mask_width: 346                   # [px]

filters:
  rf_period_same_pol: 8             # [ms] refactory period for events of same polarity
  rf_period_opp_pol: 1              # [ms] refactory period for events of opposite polarity
  nf_neighbourhood_size: 3          # [px] size of neighbourhood for neighbourhood filter
  nf_min_num_events: 3              # [] minimum number of events in neighbourhood
  nf_max_age: 70                     # [ms] maximum age of event in neighbourhood

lines:
  # promotion
  promotion_threshold: 1.2          # [px]
  promotion_num_events: 60

  # initialization
  init_bin_size: 2                  # [px]
  init_num_empty_bin: 2
  init_period: 110                  # [ms]
  init_length: 75                   # [px]

  # addition
  addition_threshold: 1.8           # [px]
  addition_mid_point_distance: 1.0

  # merge
  merge_angle_diff: deg(8.0)
  merge_distance_threshold: 3.5     # [px]
  merge_distance_multiplier: 1.4

  # hibernation
  hiberantion_newest_event_age: 25    # [ms]
  hibernation_max_hibernation_time: 200  # [ms]
  hibernation_density_threshold: 0.08  # [events / (px^2 * ms)]

  # update
  update_parameters_periodicity: 7   # [ms]
  update_parameters_new_event_ratio: 0.01
  update_parameters_num_events: 7
  update_intersection_periodicity: 3  # [ms]

  # cleanup
  cleanup_event_age_threshold: 70     # [ms]

  # deletion
  deletion_t_no_events: 100            # [ms]
  deletion_t_hibernate_no_events: 350  # [ms]
  deletion_out_of_frame_band: 3      # [px]
  deletion_min_length: 30             # [px]


clusters:
  # creation
  creation_num_events: 7

  # addition
  addition_distance_NN_threshold: 2     # [px] currently unused
  addition_distance_line_treshold: 1.3  # [px]
  addition_mid_point_threshold: 1.0     # []

  # merge
  merge_angle_diff: rad(15.0)           # [rad]

  #update
  update_parameters_num_events: 8

  #cleanup
  cleanup_event_age_threshold: 40       # [ms]

  # deletion
  deletion_t_no_events: 50              # [ms]

chains:
  # addition
  neighbourhood_size: 3
  addition_distance_threshold: 0.8

  max_chain_length: 10

  # deletion
  deletion_max_event_age: 50            # [ms]
